truncate/table_schema.go

```diff
@@ -134,6 +151,17 @@ func fetchTableSchemas(ctx context.Context, client *spanner.Client, targetTables
 		return nil, err
 	}
 
+	// Remove tables that may delete the exclude tables in cascade
+	if mode == fetchModeExclude && len(excludeParents) > 0 {
+		filtered := make([]*tableSchema, 0, len(tables))
+		for _, t := range tables {
+			if _, ok := excludeParents[t.tableName]; !ok {
+				filtered = append(filtered, t)
+			}
+		}
+		tables = filtered
+	}
```

Comment:
When in fetchModeExclude, it becomes necessary to exclude the recorded parent tables.